src/sync.rs

```diff
@@ -100,25 +110,64 @@ unsafe impl<T> Sync for GILProtected<T> where T: Send {}
 /// }
 /// # Python::with_gil(|py| assert_eq!(get_shared_list(py).len(), 0));
 /// ```
-#[derive(Default)]
-pub struct GILOnceCell<T>(UnsafeCell<Option<T>>);
+pub struct GILOnceCell<T> {
+    once: Once,
+    data: UnsafeCell<MaybeUninit<T>>,
+
+    /// (Copied from std::sync::OnceLock)
+    ///
+    /// `PhantomData` to make sure dropck understands we're dropping T in our Drop impl.
```

Comment:
```suggestion
    /// `PhantomData` to make sure drop understands we're dropping T in our Drop impl.
```

Reply:
Will leave this, `dropck` is I think short for "drop check" and comes from the standard library source I copied this from.

Reply:
Thanks!